app/models/shipit/deploy_spec/kubernetes_discovery.rb

```diff
@@ -9,24 +9,13 @@ def discover_rollback_steps
         discover_kubernetes || super
       end
 
-      def discover_machine_env
-        env = super
-        env = env.merge('K8S_TEMPLATE_FOLDER' => kube_config['template_dir']) if kube_config['template_dir']
-        env
-      end
-
       private
 
       def discover_kubernetes
         return unless kube_config.present?
 
-        [
-          Shellwords.join([
-            "kubernetes-deploy",
-            kube_config['namespace'],
-            kube_config['context'],
-          ]),
-        ]
+        template_dir = kube_config['template_dir'] || "config/deploy/$ENVIRONMENT"
+        ["kubernetes-deploy #{kube_config['namespace']} #{kube_config['context']} --template-dir #{template_dir}"]
```

Comment:
I think the problem is that `Shellwords.join` escapes the `$` in `$ENVIRONMENT`.

Using `Shellwords.escape` on the other arguments might be a good idea though.